opennmt/utils/decoding.py

```diff
@@ -427,6 +427,7 @@ def dynamic_decode(
     minimum_iterations=0,
     attention_history=False,
     attention_size=None,
+    tflite_output_size=None,
```

Comment:
I think it would be clearer to define a different decoding function that is specific for TFLite. What do you think?

Reply:
I'm unsure about this.  Initially I wrote the code defining new decoding functions.  After looking at it and realizing I could maybe reuse the existing decoding functions by adding a parameter and a few if statements, I went for it.  The reason why my preference right now is towards reusing the decoder functions is there is a lot of duplicate code otherwise.  A change in the decoding function would mean TensorFlow Lite exporting would break if a change isn't made in the TensorFlow Lite decoding function as well.  I am open to different ways of handling this

Reply:
It seems the while_loop body is still duplicated. If you can find a way to use the same loop body in both cases, then this integration looks fine. Otherwise, I would propose to define a new function `dynamic_decode_tflite` which implements a minimal greedy decoding loop: no generic `decoding_strategy object`, no attention management, no score gathering, just a plain decoding loop taking the argmax of the model output.

> A change in the decoding function would mean TensorFlow Lite exporting would break if a change isn't made in the TensorFlow Lite decoding function as well

The export should not break in this case. The graph will be different but that is expected since TFLite has limitations.

Reply:
Sounds good.  I will try reusing the while_loop body, otherwise I will create the new function.

Reply:
I have just pushed a commit addressing this which reuses the while loop!